2/index.js

```diff
@@ -1,10 +1,18 @@
-//lesson01--------------------------------------------------
-//chatGPT prompt："javascript DOM作成 li テキスト「これです」"
-//reference:https://www.wakuwakubank.com/posts/306-javascript-dom/#index_id4
-//reference:https://stackoverflow.com/questions/50250214/how-to-use-insertbefore-without-second-parameter
+//lesson02--------------------------------------------------
+//chatGPT prompt："javascript DOM作成 li aタグにテキスト挿入「これです」 aタグの中にimgタグ"
+//reference:https://developer.mozilla.org/ja/docs/Learn/JavaScript/Client-side_web_APIs/Manipulating_documents
 
 //create list item
 const List = document.getElementById('js-list'); // get:ID='js-list'
 const listItem = document.createElement('li'); // create<li></li>
-listItem.textContent = 'これです' // add:text -> これです
-List.appendChild(listItem);
\ No newline at end of file
+const listItemLink = document.createElement('a'); //create<a></a>
+const listItemLinkImg = document.createElement('img'); //create<img>
+
+listItemLink.href = "1.html";//add:linkURL
+listItemLink.textContent = 'これです' // add:text -> これです
+listItemLinkImg.src = "image/bookmark.png";//add:img -> src
+listItemLinkImg.alt = "ブックマーク";//add:img -> alt
+
+listItemLink.appendChild(listItemLinkImg);
+listItem.appendChild(listItemLink);
+List.appendChild(listItem);
```

Comment:
IMO
このように書くことができます
ref: https://developer.mozilla.org/ja/docs/Web/API/Node/appendChild#%E5%85%A5%E3%82%8C%E5%AD%90%E6%A7%8B%E9%80%A0%E3%81%AE_dom_%E3%81%AE%E4%BD%9C%E6%88%90
```suggestion
List.appendChild(listItem).appendChild(listItemLink).appendChild(listItemLinkImg);
```

Reply:
Thank you!
I wanted to know exactly this code!

Reply:
https://github.com/megusasuke0/frontendhandson/commit/f8b016bf3a2dac7a9dba279b789bf52fa82d1817